kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerServerException.java

```diff
@@ -35,25 +32,38 @@ public class SpinnakerServerException extends SpinnakerException {
    */
   private final String rawMessage;
 
+  /* RetrofitError's response body derived as map or null if response not provided in RetrofitError or custom  message has been provided*/
+  private final Map<String, Object> responseBody;
+
   /**
-   * Parses the message from the {@link RetrofitErrorResponseBody} of a {@link RetrofitError}.
+   * Parses the Error response body of {@link RetrofitError} as responseBody and extract message as
+   * rawMessage
    *
    * @param e The {@link RetrofitError} thrown by an invocation of the {@link retrofit.RestAdapter}
    */
   public SpinnakerServerException(RetrofitError e) {
     super(e.getCause());
-    RetrofitErrorResponseBody body =
-        (RetrofitErrorResponseBody) e.getBodyAs(RetrofitErrorResponseBody.class);
+    this.responseBody = (Map<String, Object>) e.getBodyAs(HashMap.class);
     this.rawMessage =
-        Optional.ofNullable(body).map(RetrofitErrorResponseBody::getMessage).orElse(e.getMessage());
+        responseBody != null
+            ? (String) responseBody.getOrDefault("message", e.getMessage())
+            : e.getMessage();
   }
 
+  /**
+   * Parses the Error response body of {@link RetrofitException} as responseBody and extract message
```

Comment:
```suggestion
   * Parse the response body of {@link RetrofitException} as json into responseBody and extract the message property
```
though as I stare at RetrofitException I wonder if it makes sense to assert that the response's errorBody isn't null...otherwise getErrorBodyAs doesn't really work properly.  I mean, it technically does, but...not really.  Sort of independent of this, it's tempting to rename RetrofitException.getErrorBodyAs to getBodyAs to hide the implementation detail of how the underlying response works.